src/modals/modal-overlay/modal-overlay.component.js

```diff
@@ -0,0 +1,136 @@
+      modalTitle: "=",
+      titleId: "=?",
+      hideCloseIcon: "<?",
+      backgroundClose: "<?",
+      onBackgroundClick: "=?",
+      isForm: "<?",
+      modalBodyTemplate: "=",
+      modalBodyScope: "=?",
+      actionButtons: "<"
+    },
+    templateUrl: 'modals/modal-overlay/modal-overlay.html',
+    controller: function ( $uibModal ) {
+      'use strict';
+
+      var ctrl = this;
+
+      ctrl.open = function () {
+        $uibModal.open({
+          component: 'pfModalOverlayContent',
+          backdrop: ctrl.backgroundClose ? true : 'static',
+          resolve: {
+            modalId: function () {
+              return ctrl.modalId;
+            },
+            titleId: function () {
+              return ctrl.titleId || "modalTitle";
+            },
+            modalTitle: function () {
+              return ctrl.modalTitle;
+            },
+            hideCloseIcon: function () {
+              return ctrl.hideCloseIcon;
+            },
+            onBackgroundClick: function() {
+              return ctrl.onBackgroundClick;
+            },
+            isForm: function() {
+              return ctrl.isForm;
+            },
+            modalBodyTemplate: function() {
+              return ctrl.modalBodyTemplate;
+            },
+            modalBodyScope: function() {
+              return ctrl.modalBodyScope;
+            },
+            actionButtons: function () {
+              return ctrl.actionButtons;
+            }
+          }
+        })
+          .result.then(
+          function (dismissCause) {
+            ctrl.close({'dismissCause': dismissCause}); // closed
+          },
+          function (dismissCause) {
+            ctrl.close({'dismissCause': dismissCause}); // dismissed
+          }
+        );
+      };
+
+      ctrl.$onInit = function () {
+        if (ctrl.showModal === undefined) {
+          ctrl.showModal = false;
+        }
+      };
+
+      ctrl.$onChanges = function (changesObj) {
+        if (changesObj.showModal && changesObj.showModal.currentValue === true) {
+          ctrl.open();
+        }
+      };
+    }
+  });
+
+
+angular.module('patternfly.modals').component('pfModalOverlayContent', {
+  templateUrl: 'modal-overlay-template.html',
+  bindings: {
+    resolve: '<',
+    close: '&',
+    dismiss: '&'
+  },
+  controller: function ($scope) {
+    'use strict';
+    var ctrl = this;
+
+    ctrl.$onInit = function () {
+      ctrl.modalId = ctrl.resolve.modalId;
+      ctrl.titleId = ctrl.resolve.titleId || "modalTitle";
+      ctrl.modalTitle = ctrl.resolve.modalTitle;
+      ctrl.hideCloseIcon = ctrl.resolve.hideCloseIcon || false;
+      ctrl.modalBodyTemplate = ctrl.resolve.modalBodyTemplate;
+      ctrl.modalBodyScope = ctrl.resolve.modalBodyScope;
+      ctrl.actionButtons = ctrl.resolve.actionButtons;
+      ctrl.isForm = ctrl.resolve.isForm;
+      ctrl.onBackgroundClick = ctrl.resolve.onBackgroundClick;
+
+      ctrl.ok = function (label, actionFn) {
```

Comment:
This can just be:
```
ctrl.ok = function (label, actionFn) {
        if (typeof actionFn === "function") {
          actionFn();
        }
        ctrl.close({$value: label});
      };
```